src/ape/types/__init__.py

```diff
@@ -143,50 +144,16 @@ def encode_topic_value(abi_type, value):
         )
 
 
-class ContractLog(BaseModel):
-    """
-    An instance of a log from a contract.
-    """
-
-    event_name: str
-    """The name of the event."""
-
-    contract_address: AddressType
-    """The contract responsible for emitting the log."""
-
-    event_arguments: Dict[str, Any]
-    """The arguments to the event, including both indexed and non-indexed data."""
-
-    transaction_hash: Any
-    """The hash of the transaction containing this log."""
-
-    block_number: int
-    """The number of the block containing the transaction that produced this log."""
-
-    block_hash: Any
-    """The hash of the block containing the transaction that produced this log."""
-
-    log_index: int
-    """The index of the log on the transaction."""
+class EventArguments(BaseModel):
 
-    transaction_index: Optional[int] = None
-    """
-    The index of the transaction's position when the log was created.
-    Is `None` when from the pending block.
-    """
+    event_arguments: Dict[str, Any] = Field(...)
 
-    @validator("block_number", "log_index", "transaction_index", pre=True)
-    def validate_hex_ints(cls, value):
-        if not isinstance(value, int):
-            return to_int(value)
-
-        return value
-
-    @validator("contract_address", pre=True)
-    def validate_address(cls, value):
-        from ape import convert
-
-        return convert(value, AddressType)
+    @cached_property
+    def topics(self):
```

Comment:
I don't see `topics` being used anywhere.

I would like to know more about why we need the data structure this way. Perhaps by seeing it in action that will clear things up